### DISCUSSION TITLE
Vocabulary Building !

### DISCUSSION THREAD

**Discussion by Rudy-alt** [xxyzz/WordDumb#256] [Q&A] [answered]
So I've noticed that when sideloading a book I have to manually setup the dictionary for it as it doesn't pick it up right up the bat, and as for function of "vocabulary building" it just says that "there are no dictionary's available in the device for the language on this book", is there a fix for this?

**Comment by xxyzz** [comment]
You should ask Amazon support...

**Reply by Rudy-alt** [reply]
I don't think it's amazon issue, because I tried it on the local dictionary and it worked fine. I read some threads of individuals having it working by converting mobi and adding the tag [EBOK] on the output section, but I haven't gotten anything working with kfx. 

Any suggestions ?

**Reply by xxyzz** [reply]
What's the book language? Do you have any dictionary for that language installed?

How is this related with this plugin? Maybe try the KFX plugin forum thread...

**Reply by xxyzz** [reply]
Non-English book will be converted to English book on Kindle to enable Word Wise, but shouldn't affect vocabulary building for English book, not sure about other language though...

**Reply by Rudy-alt** [reply]
How's that could you explain how a why a non-english book will be converted to English book?

So to sum up what I've compilled, it does work perfectly with mobi format, I managed to get it working after converting from pdf to mobi with this prompt aftter adding "[EBOK" to the personal tag, I believe maybe the same can be done to kfx, but I haven't figured how or ir you can add a tag like that that may do some changes on the book when sideloading it, if you know anything it pls let me know, it may be of use !
![Screenshot 2024-08-30 003612](https://github.com/user-attachments/assets/01914e1d-e824-4f6b-99b4-2a135641473d)

**Reply by xxyzz** [reply, marked as answer]
Because only English books can have Word Wise feature enabled on Kindle.

EBOK probably is a book type metadata, if you using this plugin to create word wise file, kfx book will always have this type because this is  required to enabled word wise.

You should ask in the [KFX plugin's thread](https://www.mobileread.com/forums/showthread.php?t=272407) about book format conversion.

I don't think vocabulary building would work for non-en book with en language metadata.

**Reply by Rudy-alt** [reply]
I guess you're right, I don't get what you mean since my spanish books have the feature enabled, but it seems to fail for vocabulary builder and setting local dictionary.
 I gave it another try this morning, I converted pdf to EPUB, and so on to KFX, and sideloaded through "Send to device", dictionary and vocabulary builder were both set up with the local dictionary. Then I tried to doing the same but for the final step instead "Send to device" I used worddumb and it gave me the same issue I've been facing, so what I get from this is maybe that by generating wordwise and X ray alters the books a bit, to where it'll give you this error.

**Reply by Rudy-alt** [reply]
My apology's I didn't understand you, so essentially why it's not working is because for word wise to be added the book need's to be converted to english, so I guess pretty much yeah, vocabulary builder won't work for non english books, very unlucky, maybe in a future version there'll be a way to get it working.
